operator/pkg/manifest/installer.go

```diff
@@ -229,10 +246,15 @@ func applyRecursive(manifests name.ManifestMap, version pkgversion.Version, opts
 			allAppliedObjects = append(allAppliedObjects, appliedObjects...)
 			mu.Unlock()
 
+			if len(componentDependencies[c]) > 0 {
+				if err := waitForResources(appliedObjects, opts); err != nil {
```

Comment:
One note is this will wait even if we don't pass `--wait`. The same is really true of the CRDs as well, so I think its fine?